core/network/src/protocol/sync.rs

```diff
@@ -118,26 +130,48 @@ pub(crate) enum AncestorSearchState<B: BlockT> {
 }
 
 #[derive(Copy, Clone, Eq, PartialEq, Debug)]
+/// The state of syncing between a Peer and ourselves. Generally two categories, "busy" or
+/// `Available`. If busy, the Enum defines what we are busy with.
 pub(crate) enum PeerSyncState<B: BlockT> {
+	/// Searching for ancestors the Peer has in common with us.
 	AncestorSearch(NumberFor<B>, AncestorSearchState<B>),
+	/// Available for sync requests.
 	Available,
+	/// Actively downloading new blocks, starting from the given Number.
 	DownloadingNew(NumberFor<B>),
+	/// Downloading a stale (already imported) block with given Hash.
```

Comment:
Stale block is a block with a number that is lower than our best number. I might be from a fork and not necessarily imported.